graph/memstore/quadstore.go

```diff
@@ -117,17 +116,17 @@ func (qs *QuadStore) ApplyDeltas(deltas []graph.Delta, ignoreOpts graph.IgnoreOp
 		case graph.Add:
 			if !ignoreOpts.IgnoreDup {
 				if _, exists := qs.indexOf(d.Quad); exists {
-					return graph.ErrQuadExists
+					return &graph.QuadError{Action: d.Action, Quad: d.Quad, Err: graph.ErrQuadExists}
```

Comment:
Ah, sorry, was thinking about other implementation.

In any case, these points remains:
a) It silently breaks callers code
b) Caller is forced to type assert AND compare instead of one compare / type assertion / calling helper
c) Generic quad error does not need to know about Delta's Action